packages/tests-postman/index.js

```diff
@@ -97,18 +103,18 @@ const referenceData = JSON.parse(fs.readFileSync(program.opts().referencedata, '
  * @throws if service provider oauth2 config is incomplete
  */
 function getOAuth2Config(name) {
-  const serv = serviceData[getServiceIdx(name)]; // eslint-disable-line no-use-before-define
+  const serv = serviceData[getServiceIdx(name)];
   if (typeof serv === 'undefined') {
     throw Error(`"${name}" is not a valid service name`);
   }
-  const { access_token_url, client_id } = serv.oauth2 || {}; // eslint-disable-line camelcase
-  if (typeof access_token_url === 'undefined') { // eslint-disable-line camelcase
+  const { access_token_url: tokenURL, client_id: clientID } = serv.oauth2 || {};
+  if (typeof tokenURL === 'undefined') {
     throw Error(`"${name}" oauth2 configuration is missing "access_token_url" value`);
   }
-  if (typeof client_id === 'undefined') { // eslint-disable-line camelcase
+  if (typeof clientID === 'undefined') {
     throw Error(`"${name}" oauth2 configuration is missing "client_id" value`);
   }
-  const key = sha256(client_id);
+  const key = sha256(clientID);
```

Comment:
clientID is public... I would prefer not to add hashing on top of identifiers that are expected to be public.
https://www.oauth.com/oauth2-servers/client-registration/client-id-secret/

> The client_id is a public identifier for apps. Even though it’s public, it’s best that it isn’t guessable by third parties, so many implementations use something like a 32-character hex string. If the client ID is guessable, it makes it slightly easier to craft phishing attacks against arbitrary applications. It must also be unique across all clients that the authorization server handles.

Seems sufficient to just store `CLIENT_ID` as a secret in CI... no need to hash it.

I would expect to be able to trivially set these values from bash without running JS code.

Reply:
I guess hidden in what I am saying is the assertion that folks passing these tests will not be publishing CLIENT_IDs.... anywhere... My understanding was that we would use CI secrets for runtime tests, and likely obtain these values directly from service providers in other cases.


Perhaps we can expand on these expectations in a call.

Reply:
The hashing here is not for security, rather it is to map the alphabet allowed for client IDs to a subset of the more restrictive alphabet allowed for GitHub secret names. This facilitates mapping of client IDs to the client secret stored in GitHub, e.g., `CLIENT_SECRET_<SHA_HASH>`. I chose client_id for this purpose because it was the only field in `service-providers.json` guaranteed to map one-to-one to the client secret.

Reply:
ugh.... ok... perhaps we can use a "string nickname" instead? I don't like the idea or computing a hash just to save a secret in github.

Reply:
We could absolutely do that instead, were you thinking along the lines of adding `oauth2.client_secret_envvar` to `service-providers.json` and then using `process.env[<client_secret_envver>]` to read the client secret from the environment?

Reply:
I will create a new ticket to discuss this